### PR TITLE
Routing to new URL, but same component

### PR DESCRIPTION
**Pull request by fabioespinosa**
Say if you have a shoe store and you are browsing a shoe, then you see a suggestion of another shoe. 

After clicking the next shoe, you might have the same component, so the component is already mounted, however you might want to log the new visit to the next shoe.

### CODE REVIEW

**Comment by timneutkens** [comment]
@vinaypuppal can you review this? 👍

File: examples/with-react-ga/components/Layout.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import React from 'react'
+import Router from 'next/router'
 import { initGA, logPageView } from '../utils/analytics'
 
+Router.onRouteChangeComplete = (url) => {
+  logPageView()
+}
+
```
Comment (vinaypuppal) [inline examples/with-react-ga/components/Layout.js]: @fabioespinosa
In next.js routing to new URL will remount this layout component (unless you are using shallow routing AFAIK) so the new pageview will be registered.

By logging pageView both in Router completion event and componentDidMount(of layout component) we may log two page views for client side routing as layout component is remounted every time the route changes. 

This is why I just used componentDidMount in this example to log pageViews which works for most of the time. For [shallow routing](https://github.com/zeit/next.js#shallow-routing) cases you may watch for route change in ` componentWillReceiveProps` and trigger pageview here.

cc @timneutkens

Comment (timneutkens) [inline examples/with-react-ga/components/Layout.js]: You're correct @vinaypuppal. It might be cleaner to use `onRouteChangeComplete` though 🤔

Comment (vinaypuppal) [inline examples/with-react-ga/components/Layout.js]: @timneutkens But how will you log page view for initial render(SSR)? As Router events are fired only on client side routing.

Comment (timneutkens) [inline examples/with-react-ga/components/Layout.js]: Very good point. Hadn't thought of that!

**Pull request closed without merging**